rpcserver/rpcserver.go

```diff
@@ -0,0 +1,60 @@
+package rpcserver
+
+import (
+	"net/http/httptest"
+
+	"github.com/gonative-cc/bitcoin-lightclient/btclightclient"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/filecoin-project/go-jsonrpc"
+	"github.com/rs/zerolog/log"
+)
+
+// Have a type with some exported methods
+type RPCServerHandler struct {
+	btcLC *btclightclient.BTCLightClient
+}
+
+func (h *RPCServerHandler) Ping(in int) int {
+	return in
+}
+
+// txn to insert bitcoin block headers to babylon chain
+func (h *RPCServerHandler) InsertHeaders(
+	blockHeaders []*wire.BlockHeader,
+) (*chainhash.Hash, error) {
+	for _, blockHeader := range blockHeaders {
+		if err := h.btcLC.InsertHeader(*blockHeader); err != nil {
+			log.Err(err).Msg("Failed to insert block header")
+
+			return nil, err
+		} else {
+			log.Info().Msgf("Inserted block header %s", blockHeader.BlockHash())
+		}
+	}
+
+	return nil, nil
```

Comment:
If you return nil for every chainHash, why don't we just return an error in this case?